src/globalshortcuts.c

```diff
@@ -0,0 +1,556 @@
+                           g_object_unref);
+
+  xdp_dbus_impl_global_shortcuts_call_list_shortcuts (impl,
+                                           request->id,
+                                           arg_session_handle,
+                                           NULL,
+                                           list_shortcuts_done,
+                                           g_object_ref (request));
+
+  xdp_dbus_global_shortcuts_complete_list_shortcuts (object, invocation, request->id);
+  return G_DBUS_METHOD_INVOCATION_HANDLED;
+}
+
+static void
+global_shortcuts_iface_init (XdpDbusGlobalShortcutsIface *iface)
+{
+  xdp_dbus_global_shortcuts_set_version (XDP_DBUS_GLOBAL_SHORTCUTS (iface), 1);
+
+  iface->handle_create_session = handle_create_session;
+  iface->handle_bind_shortcuts = handle_bind_shortcuts;
+  iface->handle_list_shortcuts = handle_list_shortcuts;
+}
+
+static void
+global_shortcuts_init (GlobalShortcuts *global_shortcuts)
+{
+  xdp_dbus_global_shortcuts_set_version(XDP_DBUS_GLOBAL_SHORTCUTS (global_shortcuts), 1);
+}
+
+static void
+global_shortcuts_class_init (GlobalShortcutsClass *klass)
+{
+  quark_request_session =
+    g_quark_from_static_string ("-xdp-request-global-shortcuts-session");
+}
+
+static void
+activated_cb (XdpDbusImplGlobalShortcuts *impl,
+                  const char *session_id,
+                  const char *shortcut_id,
+                  guint64 timestamp,
+                  gpointer data)
+{
+  g_autoptr(Session) session = lookup_session (session_id);
+  GlobalShortcutsSession *global_shortcuts_session = (GlobalShortcutsSession *)session;
+
+  g_debug ("Received activated %s for %s", session_id, shortcut_id);
+
+  if (global_shortcuts_session && !global_shortcuts_session->closed) {
+    xdp_dbus_global_shortcuts_emit_activated(data, session_id, shortcut_id, timestamp);
+  }
+}
+
+static void
+deactivated_cb (XdpDbusImplGlobalShortcuts *impl,
+                  const char *session_id,
+                  const char *shortcut_id,
+                  guint64 timestamp,
+                  gpointer data)
+{
+  g_autoptr(Session) session = lookup_session (session_id);
+  GlobalShortcutsSession *global_shortcuts_session = (GlobalShortcutsSession *)session;
+
+  g_debug ("Received deactivated %s for %s", session_id, shortcut_id);
+
+  if (global_shortcuts_session && !global_shortcuts_session->closed) {
+    xdp_dbus_global_shortcuts_emit_deactivated(data, session_id, shortcut_id, timestamp);
+  }
+}
+
+static void
+shortcutschanged_cb (XdpDbusImplGlobalShortcuts *impl,
+                  const char *session_id,
+                  gpointer data)
+{
+  g_autoptr(Session) session = lookup_session (session_id);
+  GlobalShortcutsSession *global_shortcuts_session = (GlobalShortcutsSession *)session;
+
+  g_debug ("Received ShortcutsChanged %s", session_id);
+
+  if (global_shortcuts_session && !global_shortcuts_session->closed) {
+    xdp_dbus_global_shortcuts_emit_shortcuts_changed(data, session_id);
```

Comment:
All function calls should have a space before the parenthesis (e.g. `xdp_dbus_global_shortcuts_emit_shortcuts_changed (`)